lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -292,7 +318,46 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
                             HttpStatus.SC_INTERNAL_SERVER_ERROR,
                             ErrorResponse.FAILED_TO_PARSE_SUCCESSFUL_VC_STORE_ITEMS)
                     .toObjectMap();
+        } catch (ConfigException e) {
+            LOGGER.error("Configuration error", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.FAILED_TO_PARSE_CONFIG)
+                    .toObjectMap();
+        } catch (UnrecognisedCiException e) {
+            LOGGER.error("Unrecognised CI code received", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.UNRECOGNISED_CI_CODE)
+                    .toObjectMap();
+        }
+    }
+
+    private Optional<JourneyResponse> checkCiScoring(
+            IpvSessionItem ipvSessionItem,
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            String govukSigninJourneyId,
+            String ipAddress)
+            throws CiRetrievalException, UnrecognisedCiException, ConfigException {
+        final Optional<JourneyResponse> contraIndicatorJourneyResponse =
+                gpg45ProfileEvaluator.getJourneyResponseForStoredContraIndicators(
+                        ciMitService.getContraIndicatorsVC(
+                                clientOAuthSessionItem.getUserId(),
+                                govukSigninJourneyId,
+                                ipAddress),
+                        ipvSessionItem);
+
+        if (contraIndicatorJourneyResponse.isPresent()) {
+            StringMapMessage message = new StringMapMessage();
+            JourneyResponse journeyResponse = contraIndicatorJourneyResponse.get();
+            message.with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "Returning CI error response.")
+                    .with(LOG_ERROR_JOURNEY_RESPONSE.getFieldName(), journeyResponse.toString());
+            LOGGER.info(message);
+            return Optional.of(journeyResponse);
```

Comment:
You could do this slightly more neatly by using `Optional#ifPresent`:
```
contraIndicatorJourneyResponse.ifPresent((journeyResponse) -> {
    StringMapMessage message = new StringMapMessage();
    message.with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "Returning CI error response.")
            .with(LOG_ERROR_JOURNEY_RESPONSE.getFieldName(), journeyResponse.toString());
    LOGGER.info(message);
});
```

Reply:
done